src/app/services/article.service.ts

```diff
@@ -53,28 +53,22 @@ export class ArticleService {
     docRef.set({ path: fullPath, downloadUrl: url });
   }
 
-  async getLatestArticles() {
-    const articlesRef = this.fsdb.collection('articleData/articles/articles');
-    const querySnap = await articlesRef.orderBy('timestamp', 'desc').limit(12).get();
-    return this.arrayFromCollectionSnapshot(querySnap);
+  async latestArticlesRef() {
+    return this.fsdb.collection('articleData/articles/articles', ref => ref.orderBy('timestamp', 'desc').limit(12)).valueChanges();
   }
 
-
-  async getAllArticles() {
-    const articlesRef = this.fsdb.collection('articleData/articles/articles');
-    const querySnap = await articlesRef.get();
-    const articleArray = this.arrayFromCollectionSnapshot(querySnap);
-
-    return articleArray;
+  async allArticlesRef() {
+    return this.fsdb.collection('articleData/articles/articles', ref => ref.orderBy('lastUpdated', 'desc')).valueChanges();
   }
 
 
   watchBookmarkedArticles(userKey) {
-    const bookmarksRef = this.rtdb.ref(`userInfo/articleBookmarksPerUser/${userKey}`);
-    bookmarksRef.on('value', articleIds => {
+    const bookmarksRef = this.rtdb.list(`userInfo/articleBookmarksPerUser/${userKey}`).snapshotChanges();
+    bookmarksRef.subscribe(articleIds => {
       const articlesList = new Array<any>();
-      articleIds.forEach(key => {
-        this.getArticleById(key.key).then(article => {
+      articleIds.forEach(async key => {
```

Comment:
And drop the again-outdated async/await syntax
```suggestion
      articleIds.forEach(key => {
```